core/src/main/java/io/cloudevents/core/impl/BaseCloudEvent.java

```diff
@@ -65,20 +65,14 @@ public <T extends CloudEventWriter<V>, V> V read(CloudEventWriterFactory<T, V> w
     }
 
     protected void readExtensions(CloudEventContextWriter writer) throws CloudEventRWException {
-        // TODO to be improved
         for (Map.Entry<String, Object> entry : this.extensions.entrySet()) {
-            if (entry.getValue() instanceof String) {
-                writer.withContextAttribute(entry.getKey(), (String) entry.getValue());
-            } else if (entry.getValue() instanceof Number) {
-                writer.withContextAttribute(entry.getKey(), (Number) entry.getValue());
-            } else if (entry.getValue() instanceof Boolean) {
-                writer.withContextAttribute(entry.getKey(), (Boolean) entry.getValue());
-            } else if (entry.getValue() instanceof URI) {
-                writer.withContextAttribute(entry.getKey(), (URI) entry.getValue());
-            } else if (entry.getValue() instanceof OffsetDateTime) {
-                writer.withContextAttribute(entry.getKey(), (OffsetDateTime) entry.getValue());
-            } else if (entry.getValue() instanceof byte[]) {
-                writer.withContextAttribute(entry.getKey(), (byte[]) entry.getValue());
+            Object value = entry.getValue();
+            if (value instanceof String || value instanceof Number || value instanceof Boolean || value instanceof URI) {
+                writer.withContextAttribute(entry.getKey(), value.toString());
```

Comment:
Using the `.toString()` is not equivalent to previous implementation that was using typed methods.

The original issue is not clear what the optimization is?